Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs

```diff
@@ -1,182 +1,182 @@
-            var action = Substitute.For<ICauseAnarchy>();
-            var schedule = new Schedule();
-            var timer = Substitute.For<IHandleTime>();
-            var factory = new AnarchyActionFactory(action, timer);
-
-            //Act
-            Action sutCall = () => factory.WithSchedule(schedule);
-
-            //Assert
-            sutCall.Should().Throw<UnschedulableActionException>();
-        }
-    }
-}
+﻿using System;
+using System.Threading;
+using FluentAssertions;
+using Just.Anarchy.Exceptions;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Just.Anarchy.Test.Unit.Actions
+{
+    [TestFixture]
+    public class AnarchyActionFactoryTests
+    {
+        [Test]
+        public void ForTargetPatternSetsTargetPatternIfNotWhitespace()
+        {
+            //Arrange
+            var action = Substitute.For<ICauseAnarchy>();
+            var timer = Substitute.For<IHandleTime>();
+            var sut = new AnarchyActionFactory(action, timer);
+            var targetPattern = ".*";
+            //Act
+            sut.ForTargetPattern(targetPattern);
+
+            //Assert
+            sut.TargetPattern.Should().Be(targetPattern);
+        }
+
+        [Test]
+        [TestCase("\t")]
+        [TestCase("\r\n")]
+        [TestCase("")]
+        public void ForTargetPatternThrowsIfWhitespace(string targetPattern)
+        {
+            //Arrange
+            var action = Substitute.For<ICauseAnarchy>();
+            var timer = Substitute.For<IHandleTime>();
+            var sut = new AnarchyActionFactory(action, timer);
+            Action forTargetPattern = () => sut.ForTargetPattern(targetPattern);
+
+            //Act
+            forTargetPattern.Should().Throw<ArgumentException>();
+            
+            //Assert
+            action.DidNotReceive().ExecuteAsync(Arg.Any<TimeSpan?>(), Arg.Any<CancellationToken>());
+        }
+
+        [Test]
+        public void ForTargetPatternNullDisablesHandleRequest()
+        {
+            //Arrange
+            var action = Substitute.For<ICauseAnarchy>();
+            var timer = Substitute.For<IHandleTime>();
+            var sut = new AnarchyActionFactory(action, timer);
+            sut.ForTargetPattern(null);
+
+            //Act
+            sut.HandleRequest(null);
+
+            //Assert
+            action.DidNotReceive().ExecuteAsync(Arg.Any<TimeSpan?>(), Arg.Any<CancellationToken>()); ;
+        }
+
+        [Test]
+        [TestCase("/bob", true)]
+        [TestCase("/jim", false)]
+        public void HandleRequestRejectsIfTargetPatternDoesNotMatch(string url, bool expMatch)
+        {
+            //Arrange
+            var action = Substitute.For<ICauseAnarchy>();
+            var timer = Substitute.For<IHandleTime>();
+            var sut = new AnarchyActionFactory(action, timer);
+            sut.ForTargetPattern("/bob$");
+
+            //Act
+            sut.HandleRequest(url);
+
+            //Assert
+            if (expMatch) {
```

Comment:
Do you think its worth splitting this into two separate tests? So we don't need a conditional.

Reply:
Fair point, originally I think I was going to have more cases but pared it down again, splitting into two tests.